### ISSUE TITLE
Integration Test Fails on GPU

### ISSUE THREAD

**Issue by ruyimarone** [IBM/pytorch-seq2seq#120]
While investigating proper usage of the TopKDecoder, I found that the tests in `scripts/integration_test.sh` fail when running on a GPU enabled machine with the following error:
```
Traceback (most recent call last):
  File "scripts/integration_test.py", line 116, in <module>
    seq = predictor.predict(inp_seq.split())
  File "/home/marc/endtoend/pytorch-seq2seq/seq2seq/evaluator/predictor.py", line 40, in predict
    softmax_list, _, other = self.model(src_id_seq, [len(src_seq)])
  File "/usr/local/lib/python2.7/dist-packages/torch/nn/modules/module.py", line 325, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/marc/endtoend/pytorch-seq2seq/seq2seq/models/seq2seq.py", line 53, in forward
    teacher_forcing_ratio=teacher_forcing_ratio)
  File "/usr/local/lib/python2.7/dist-packages/torch/nn/modules/module.py", line 325, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/marc/endtoend/pytorch-seq2seq/seq2seq/models/TopKDecoder.py", line 131, in forward
    inflated_encoder_outputs, function=function)
  File "/home/marc/endtoend/pytorch-seq2seq/seq2seq/models/DecoderRNN.py", line 96, in forward_step
    embedded = self.embedding(input_var)
  File "/usr/local/lib/python2.7/dist-packages/torch/nn/modules/module.py", line 325, in __call__
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/torch/nn/modules/sparse.py", line 103, in forward
    self.scale_grad_by_freq, self.sparse
  File "/usr/local/lib/python2.7/dist-packages/torch/nn/_functions/thnn/sparse.py", line 59, in forward
    output = torch.index_select(weight, 0, indices.view(-1))
TypeError: torch.index_select received an invalid combination of arguments - got (torch.cuda.FloatTensor, int, torch.LongTensor), but expected (torch.cuda.FloatTensor source, int dim, torch.cuda.LongTensor index)
```

I've confirmed that this error does not occur on another machine without a GPU. Perhaps #102 is related? I can provide version and hardware information if necessary. Any help would be appreciated, thanks!

**Comment by talmachani** [comment]
I'm having the same issue...
I've got it to work, I'm guessing you are using LSTM EncoderDecoder.

see my comment [here](https://github.com/IBM/pytorch-seq2seq/pull/123)

**Comment by pskrunner14** [comment]
Fixed after #161
